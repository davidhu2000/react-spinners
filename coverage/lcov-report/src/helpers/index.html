
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    