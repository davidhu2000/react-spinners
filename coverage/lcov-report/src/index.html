
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> src</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.57% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>1431/1776</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.5% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1133/1654</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">74.9% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>206/275</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.45% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>947/1177</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="BarLoader.tsx"><a href="BarLoader.tsx.html">BarLoader.tsx</a></td>
	<td data-value="79.72" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.72" class="pct medium">79.72%</td>
	<td data-value="74" class="abs medium">59/74</td>
	<td data-value="70.12" class="pct medium">70.12%</td>
	<td data-value="77" class="abs medium">54/77</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.59" class="pct medium">79.59%</td>
	<td data-value="49" class="abs medium">39/49</td>
	</tr>

<tr>
	<td class="file medium" data-value="BeatLoader.tsx"><a href="BeatLoader.tsx.html">BeatLoader.tsx</a></td>
	<td data-value="79.16" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="72" class="abs medium">57/72</td>
	<td data-value="68.49" class="pct medium">68.49%</td>
	<td data-value="73" class="abs medium">50/73</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="78.72" class="pct medium">78.72%</td>
	<td data-value="47" class="abs medium">37/47</td>
	</tr>

<tr>
	<td class="file medium" data-value="BounceLoader.tsx"><a href="BounceLoader.tsx.html">BounceLoader.tsx</a></td>
	<td data-value="78.87" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.87" class="pct medium">78.87%</td>
	<td data-value="71" class="abs medium">56/71</td>
	<td data-value="69.01" class="pct medium">69.01%</td>
	<td data-value="71" class="abs medium">49/71</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="48" class="abs medium">38/48</td>
	</tr>

<tr>
	<td class="file medium" data-value="CircleLoader.tsx"><a href="CircleLoader.tsx.html">CircleLoader.tsx</a></td>
	<td data-value="79.45" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.45" class="pct medium">79.45%</td>
	<td data-value="73" class="abs medium">58/73</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="48" class="abs medium">38/48</td>
	</tr>

<tr>
	<td class="file medium" data-value="ClimbingBoxLoader.tsx"><a href="ClimbingBoxLoader.tsx.html">ClimbingBoxLoader.tsx</a></td>
	<td data-value="78.87" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.87" class="pct medium">78.87%</td>
	<td data-value="71" class="abs medium">56/71</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="48" class="abs medium">38/48</td>
	</tr>

<tr>
	<td class="file medium" data-value="ClipLoader.tsx"><a href="ClipLoader.tsx.html">ClipLoader.tsx</a></td>
	<td data-value="77.94" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 77%"></div><div class="cover-empty" style="width: 23%"></div></div>
	</td>
	<td data-value="77.94" class="pct medium">77.94%</td>
	<td data-value="68" class="abs medium">53/68</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="45" class="abs medium">35/45</td>
	</tr>

<tr>
	<td class="file medium" data-value="ClockLoader.tsx"><a href="ClockLoader.tsx.html">ClockLoader.tsx</a></td>
	<td data-value="79.45" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.45" class="pct medium">79.45%</td>
	<td data-value="73" class="abs medium">58/73</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="48" class="abs medium">38/48</td>
	</tr>

<tr>
	<td class="file medium" data-value="DotLoader.tsx"><a href="DotLoader.tsx.html">DotLoader.tsx</a></td>
	<td data-value="79.72" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.72" class="pct medium">79.72%</td>
	<td data-value="74" class="abs medium">59/74</td>
	<td data-value="70.66" class="pct medium">70.66%</td>
	<td data-value="75" class="abs medium">53/75</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.59" class="pct medium">79.59%</td>
	<td data-value="49" class="abs medium">39/49</td>
	</tr>

<tr>
	<td class="file high" data-value="FadeLoader.tsx"><a href="FadeLoader.tsx.html">FadeLoader.tsx</a></td>
	<td data-value="82.75" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 82%"></div><div class="cover-empty" style="width: 18%"></div></div>
	</td>
	<td data-value="82.75" class="pct high">82.75%</td>
	<td data-value="87" class="abs high">72/87</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="75" class="abs medium">50/75</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="82.75" class="pct high">82.75%</td>
	<td data-value="58" class="abs high">48/58</td>
	</tr>

<tr>
	<td class="file high" data-value="GridLoader.tsx"><a href="GridLoader.tsx.html">GridLoader.tsx</a></td>
	<td data-value="81.01" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 81%"></div><div class="cover-empty" style="width: 19%"></div></div>
	</td>
	<td data-value="81.01" class="pct high">81.01%</td>
	<td data-value="79" class="abs high">64/79</td>
	<td data-value="68.49" class="pct medium">68.49%</td>
	<td data-value="73" class="abs medium">50/73</td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="13" class="abs medium">10/13</td>
	<td data-value="81.13" class="pct high">81.13%</td>
	<td data-value="53" class="abs high">43/53</td>
	</tr>

<tr>
	<td class="file high" data-value="HashLoader.tsx"><a href="HashLoader.tsx.html">HashLoader.tsx</a></td>
	<td data-value="81.01" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 81%"></div><div class="cover-empty" style="width: 19%"></div></div>
	</td>
	<td data-value="81.01" class="pct high">81.01%</td>
	<td data-value="79" class="abs high">64/79</td>
	<td data-value="69.01" class="pct medium">69.01%</td>
	<td data-value="71" class="abs medium">49/71</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="81.48" class="pct high">81.48%</td>
	<td data-value="54" class="abs high">44/54</td>
	</tr>

<tr>
	<td class="file high" data-value="MoonLoader.tsx"><a href="MoonLoader.tsx.html">MoonLoader.tsx</a></td>
	<td data-value="80.26" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80.26" class="pct high">80.26%</td>
	<td data-value="76" class="abs high">61/76</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="80.39" class="pct high">80.39%</td>
	<td data-value="51" class="abs high">41/51</td>
	</tr>

<tr>
	<td class="file high" data-value="PacmanLoader.tsx"><a href="PacmanLoader.tsx.html">PacmanLoader.tsx</a></td>
	<td data-value="81.7" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 81%"></div><div class="cover-empty" style="width: 19%"></div></div>
	</td>
	<td data-value="81.7" class="pct high">81.7%</td>
	<td data-value="82" class="abs high">67/82</td>
	<td data-value="69.33" class="pct medium">69.33%</td>
	<td data-value="75" class="abs medium">52/75</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="12" class="abs medium">9/12</td>
	<td data-value="81.81" class="pct high">81.81%</td>
	<td data-value="55" class="abs high">45/55</td>
	</tr>

<tr>
	<td class="file medium" data-value="PropagateLoader.tsx"><a href="PropagateLoader.tsx.html">PropagateLoader.tsx</a></td>
	<td data-value="79.72" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.72" class="pct medium">79.72%</td>
	<td data-value="74" class="abs medium">59/74</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.59" class="pct medium">79.59%</td>
	<td data-value="49" class="abs medium">39/49</td>
	</tr>

<tr>
	<td class="file medium" data-value="PuffLoader.tsx"><a href="PuffLoader.tsx.html">PuffLoader.tsx</a></td>
	<td data-value="78.57" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="70" class="abs medium">55/70</td>
	<td data-value="69.01" class="pct medium">69.01%</td>
	<td data-value="71" class="abs medium">49/71</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="78.72" class="pct medium">78.72%</td>
	<td data-value="47" class="abs medium">37/47</td>
	</tr>

<tr>
	<td class="file medium" data-value="PulseLoader.tsx"><a href="PulseLoader.tsx.html">PulseLoader.tsx</a></td>
	<td data-value="79.16" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="72" class="abs medium">57/72</td>
	<td data-value="67.6" class="pct medium">67.6%</td>
	<td data-value="71" class="abs medium">48/71</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="78.72" class="pct medium">78.72%</td>
	<td data-value="47" class="abs medium">37/47</td>
	</tr>

<tr>
	<td class="file medium" data-value="RingLoader.tsx"><a href="RingLoader.tsx.html">RingLoader.tsx</a></td>
	<td data-value="79.72" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.72" class="pct medium">79.72%</td>
	<td data-value="74" class="abs medium">59/74</td>
	<td data-value="69.01" class="pct medium">69.01%</td>
	<td data-value="71" class="abs medium">49/71</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.59" class="pct medium">79.59%</td>
	<td data-value="49" class="abs medium">39/49</td>
	</tr>

<tr>
	<td class="file medium" data-value="RiseLoader.tsx"><a href="RiseLoader.tsx.html">RiseLoader.tsx</a></td>
	<td data-value="79.45" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.45" class="pct medium">79.45%</td>
	<td data-value="73" class="abs medium">58/73</td>
	<td data-value="68.49" class="pct medium">68.49%</td>
	<td data-value="73" class="abs medium">50/73</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="48" class="abs medium">38/48</td>
	</tr>

<tr>
	<td class="file high" data-value="RotateLoader.tsx"><a href="RotateLoader.tsx.html">RotateLoader.tsx</a></td>
	<td data-value="80.51" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80.51" class="pct high">80.51%</td>
	<td data-value="77" class="abs high">62/77</td>
	<td data-value="68.49" class="pct medium">68.49%</td>
	<td data-value="73" class="abs medium">50/73</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="50" class="abs high">40/50</td>
	</tr>

<tr>
	<td class="file high" data-value="ScaleLoader.tsx"><a href="ScaleLoader.tsx.html">ScaleLoader.tsx</a></td>
	<td data-value="80.26" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80.26" class="pct high">80.26%</td>
	<td data-value="76" class="abs high">61/76</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="75" class="abs medium">50/75</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="78.72" class="pct medium">78.72%</td>
	<td data-value="47" class="abs medium">37/47</td>
	</tr>

<tr>
	<td class="file medium" data-value="SkewLoader.tsx"><a href="SkewLoader.tsx.html">SkewLoader.tsx</a></td>
	<td data-value="77.94" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 77%"></div><div class="cover-empty" style="width: 23%"></div></div>
	</td>
	<td data-value="77.94" class="pct medium">77.94%</td>
	<td data-value="68" class="abs medium">53/68</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="45" class="abs medium">35/45</td>
	</tr>

<tr>
	<td class="file medium" data-value="SquareLoader.tsx"><a href="SquareLoader.tsx.html">SquareLoader.tsx</a></td>
	<td data-value="77.94" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 77%"></div><div class="cover-empty" style="width: 23%"></div></div>
	</td>
	<td data-value="77.94" class="pct medium">77.94%</td>
	<td data-value="68" class="abs medium">53/68</td>
	<td data-value="68.11" class="pct medium">68.11%</td>
	<td data-value="69" class="abs medium">47/69</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="45" class="abs medium">35/45</td>
	</tr>

<tr>
	<td class="file medium" data-value="SyncLoader.tsx"><a href="SyncLoader.tsx.html">SyncLoader.tsx</a></td>
	<td data-value="79.16" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="72" class="abs medium">57/72</td>
	<td data-value="67.6" class="pct medium">67.6%</td>
	<td data-value="71" class="abs medium">48/71</td>
	<td data-value="72.72" class="pct medium">72.72%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="78.72" class="pct medium">78.72%</td>
	<td data-value="47" class="abs medium">37/47</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="73" class="abs high">73/73</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="50" class="abs high">50/50</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    