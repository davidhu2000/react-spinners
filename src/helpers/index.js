export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
