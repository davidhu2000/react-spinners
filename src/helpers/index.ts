export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
