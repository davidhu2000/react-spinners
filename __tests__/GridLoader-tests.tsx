import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from "jest-emotion";
expect.extend(matchers);

import GridLoader from "../src/GridLoader";
import { LoaderSizeMarginProps } from "../src/interfaces";
import { sizeMarginDefaults } from "../src/helpers";

interface MockMath {
  random: () => number;
}

const mockMath: MockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath as Math;

describe("GridLoader", () => {
  let loader: ReactWrapper;
  let props: LoaderSizeMarginProps;
  const defaultColor = "#000000";
  const defaultSize = 15;
  const defaultUnit = "px";
  const wrapperWidthDefault = parseFloat(defaultSize.toString()) * 3 + parseFloat("2px") * 6;

  it("should match snapshot", () => {
    loader = mount(<GridLoader />);
    expect(loader).toMatchSnapshot();
  });

  it("should contain default props if no props are passed", () => {
    props = loader.props();
    expect(props).toEqual(sizeMarginDefaults(defaultSize));
  });

  it("should contain styles created using default props", () => {
    expect(loader).toHaveStyleRule("width", `${wrapperWidthDefault}${defaultUnit}`);

    for (let i = 0; i < 9; i++) {
      expect(loader.find("span span").at(i)).toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "height",
        `${defaultSize}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${defaultSize}${defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule("margin", "2px");
    }
  });

  it("should render null if loading prop is set as false", () => {
    loader = mount(<GridLoader loading={false} />);
    expect(loader.isEmptyRender()).toBe(true);
  });

  it("should render the correct color based on prop", () => {
    const color = "#e2e2e2";
    loader = mount(<GridLoader color={color} />);
    expect(loader.find("span span")).not.toHaveStyleRule("background-color", defaultColor);
    expect(loader.find("span span")).toHaveStyleRule("background-color", color);
  });

  it("should render the correct size for the parent span based on props", () => {
    const size = 18;
    const wrapperWidth = parseFloat(size.toString()) * 3 + parseFloat("2px") * 6;

    loader = mount(<GridLoader size={size} />);
    expect(loader).not.toHaveStyleRule("width", `${wrapperWidthDefault}${defaultUnit}`);
    expect(loader).toHaveStyleRule("width", `${wrapperWidth}${defaultUnit}`);

    for (let i = 0; i < 9; i++) {
      expect(loader.find("span span").at(i)).not.toHaveStyleRule(
        "height",
        `${defaultSize}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).not.toHaveStyleRule(
        "width",
        `${defaultSize}${defaultUnit}`
      );

      expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${size}${defaultUnit}`);
      expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${size}${defaultUnit}`);
    }
  });

  it("should render the css override based on props", () => {
    loader = mount(
      <GridLoader css={"font-size: 11px; width: 100px; height: 200px; color: blue;"} />
    );
    expect(loader).not.toHaveStyleRule("font-size", "0");
    expect(loader).not.toHaveStyleRule("width", "57px");
    expect(loader).toHaveStyleRule("font-size", "11px");
    expect(loader).toHaveStyleRule("width", "100px");
  });
});
