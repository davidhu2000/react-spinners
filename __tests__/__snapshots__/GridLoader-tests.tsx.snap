// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridLoader should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    -webkit-transform: scale(0.5);
    -ms-transform: scale(0.5);
    transform: scale(0.5);
    opacity: 0.7;
  }

  100% {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
    opacity: 1;
  }
}

.emotion-9 {
  width: 57px;
  font-size: 0;
}

.emotion-0 {
  display: inline-block;
  background-color: #000000;
  width: 15px;
  height: 15px;
  margin: 2px;
  border-radius: 100%;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.1s 0.3s infinite ease;
  animation: animation-0 1.1s 0.3s infinite ease;
}

<onlyUpdateForKeys(Loader)
  color="#000000"
  css={Object {}}
  loading={true}
  margin="2px"
  size={15}
  sizeUnit="px"
>
  <Loader
    color="#000000"
    css={Object {}}
    loading={true}
    margin="2px"
    size={15}
    sizeUnit="px"
  >
    <div
      className="emotion-9"
    >
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
      <div
        className="emotion-0"
      />
    </div>
  </Loader>
</onlyUpdateForKeys(Loader)>
`;
